internal/client/client.go

```diff
@@ -167,8 +167,7 @@ func (tsc *ThriftServiceClient) ExecuteStatement(ctx context.Context, req *cli_s
 	log, ctx = LoggerAndContext(ctx, req)
 	msg, start := log.Track("ExecuteStatement")
 
-	// We use context.Background to fix a problem where on context done the query would not be cancelled.
-	resp, err := tsc.TCLIServiceClient.ExecuteStatement(context.Background(), req)
+	resp, err := tsc.TCLIServiceClient.ExecuteStatement(ctx, req)
```

Comment:
@andrefurlan-db @rcypher-databricks I need your help here. It's unclear to me what (and how) `context.Background()` should fix, and the comment actually doesn't help a lot. If there was any issue with using `ctx` itself - I would like to know what it was, and how can I check if it's still there. Thanks!

Reply:
@kravets-levko any progress in here getting this merged? any way i can help?

Reply:
@shelldandy sorry for the delay. I did try the library a lot with both variants, and haven't notice any differences in behavior. Maybe, `ctx.Background` was needed to fix some very specific case, but I don't know what it is. I tend to merge your fix, just maybe will check it once more just to make sure I don't miss something obvious

Reply:
@kravets-levko FWIW we've been using my fork for a while without problems on production so I'm also inclined to think that whatever the reason for the background use was has already been fixed

Reply:
@kravets-levko bumping again to see if we can get this one merged. its needed for use with a custom authenticator that needs the `ctx` to get some values related to the authentication flow